pkg/conversion/decode.go

```diff
@@ -80,6 +80,17 @@ func (s *Scheme) Decode(data []byte) (interface{}, error) {
 // If obj's version doesn't match that in data, an attempt will be made to convert
 // data into obj's version.
 func (s *Scheme) DecodeInto(data []byte, obj interface{}) error {
+	return s.DecodeIntoWithSpecifiedVersionKind(data, obj, "", "")
+}
+
+//DecodeIntoWithSpecifiedVersionKind compares the passed in specifiedVersion and
+//specifiedKind with data.Version and data.Kind, defaulting data.Version and
+//data.Kind to the specified value if they are empty, or generating an error if
+//data.Version and data.Kind are not empty and differ from the specified value.
+//The function then implements the functionality of DecodeInto.
+//If specifiedVersion and specifiedKind are empty, the function degenerates to
+//DecodeInto.
+func (s *Scheme) DecodeIntoWithSpecifiedVersionKind(data []byte, obj interface{}, specifiedVersion, specifiedKind string) error {
```

Comment:
nit: We put a space between the comment and "//". So the comments should be
```
// DecodeInto...
```
instead of 
```
//DecodeInto...
```